recipe-server/client/control/state/actions/actions.js

```diff
@@ -0,0 +1,43 @@
+import {
+  ACTION_RECEIVE,
+} from '../action-types';
+
+import {
+  makeApiRequest,
+} from '../requests/actions';
+
+
+function receiveAction(action) {
+  return dispatch => {
+    dispatch({
+      type: ACTION_RECEIVE,
+      action,
+    });
+  };
+}
+
+
+export function fetchAction(name) {
+  const requestId = `fetch-${name}`;
+  return makeApiRequest(requestId, `action/${name}/`, { method: 'GET' })
+    .then(action => receiveAction(action));
```

Comment:
I'm actually fine with abstracting into its own function - skimming the code, it's faster for me to parse what's going on, and enforces separation of concerns. I prefer the 'redirect' to another method over reading a few lines of an inlined function to determine what it's actually doing, but maybe that's just me. Also, refactoring or adding to `recieveAction` would be cleaner if we ever decided to do so in the future.

But, the line itself could be rewritten to just `.then(receiveAction)`, which I would suggest.